ledger/src/blockstore_db.rs

```diff
@@ -1418,16 +1420,26 @@ impl WriteBatch {
     }
 }
 
-impl<C> LedgerColumn<C>
+impl<C, const K: usize> LedgerColumn<C, K>
```

Comment:
does it work to use `C::KEY_LEN` rather than passing it in here?

Reply:
Great question - I meant to include a note about this but it has been a month since I initially posted this and forgot 🙈 

Unfortunately, we cannot. Here is a little sample without the extra generic parameter `K`:
```rust
pub struct LedgerColumnTest<C>
where
    C: Column + ColumnName,
{
    column: PhantomData<C>,
}

impl<C> LedgerColumnTest<C>
where
    C: Column + ColumnName,
{
    fn key_from_index(index: C::Index) -> [u8; C::KEY_LEN] {
        let mut key = [0u8; C::KEY_LEN];
        C::serialize_index(&mut key, index);
        key
    }
}
```
which fails to compile with:
```
error: generic parameters may not be used in const operations
    --> ledger/src/blockstore_db.rs:1363:48
     |
1363 |     fn key_from_index(index: C::Index) -> [u8; C::KEY_LEN] {
     |                                                ^^^^^^^^^^ cannot perform const operation using `C`
     |
     = note: type parameters may not be used in const expressions

error: constant expression depends on a generic parameter
    --> ledger/src/blockstore_db.rs:1364:29
     |
1364 |         let mut key = [0u8; C::KEY_LEN];
     |                             ^^^^^^^^^^
     |
     = note: this may fail depending on what value the parameter takes

error: could not compile `solana-ledger` (lib) due to 2 previous errors
```
If/when the following feature stabilizes, I think we could move that direction:
https://doc.rust-lang.org/beta/unstable-book/language-features/generic-const-exprs.html

Reply:
Mentioned this item in the PR description so marking this as resolved